output.go

```diff
@@ -216,6 +216,8 @@ func (c *Output) writeFormatter(ctx *Context, formatter Formatter, value interfa
 	if newline {
 		fmt.Fprintln(target)
 	}
+	// supresses machine error to stdout
+	ctx.suppressMachineError = true
```

Comment:
When is supressMachineError false? It's looks like it's hard coded here to true but I assume it's  used as false elsewhere.

Reply:
It defaults to false, essentially if the command writes to stdout we don't need to output anything to stdout since the command has already output something.

Reply:
I've adjusted to make it clearer